src/components/product/product-context.js

```diff
@@ -0,0 +1,34 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import axios from "axios";
+
+const ProductContext = createContext();
+
+export function ProductProvider({ children }) {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    (async function () {
+      try {
+        const { status, data } = await axios.get(
+          `${process.env.REACT_APP_BACKEND_BASE_URL}/product`
+        );
+        if (status === 200) {
+          setProducts(() => [...data.products]);
+        }
+      } catch (error) {
+        console.log(error);
```

Comment:
Use better Error management every place

Reply:
will do that for sure in the upcoming update